system/firewalld.py

```diff
@@ -104,15 +104,49 @@
 - firewalld: masquerade=yes state=enabled permanent=true zone=dmz
 '''
 
+import sys
+
+#####################
+# Globals
+#
 fw = None
+module = None
 fw_offline = False
 Rich_Rule = None
 FirewallClientZoneSettings = None
 
 #####################
-# fw_offline helpers
+# exception handling
 #
+def action_handler(action_func, action_func_args):
+    """
+    Function to wrap calls to make actions on firewalld in try/except
+    logic and emit (hopefully) useful error messages
+    """
+
+    msgs = []
+
+    try:
+        return action_func(*action_func_args)
+    except Exception:
+        # Make python 2.4 shippable ci tests happy
+        e = sys.exc_info()[1]
```

Comment:
Can use our helper function so people don't have to remember what sys.exc_info[1] means:
```
from ansible.module_utils.pycompat24 import get_exception
[...]
e = get_exception()
```

It does the same thing.